lib/new_relic/agent/instrumentation/opensearch/instrumentation.rb

```diff
@@ -0,0 +1,66 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic::Agent::Instrumentation
+  module OpenSearch
+    PRODUCT_NAME = 'OpenSearch'
+    OPERATION = 'perform_request'
+    OPERATION_PATTERN = %r{/lib/opensearch/api/(?!.+#{OPERATION})}
+    INSTANCE_METHOD_PATTERN = /:in (?:`|')(?:.+#)?([^']+)'\z/
+    INSTRUMENTATION_NAME = NewRelic::Agent.base_name(name)
+
+    def perform_request_with_tracing(_method, _path, params = {}, body = nil, _headers = nil, _opts = {}, &_block)
```

Comment:
I'm looking at the two Hash instantiations happening per call when the defaults are involved.

- Perhaps the `params` one can be set to `NewRelic::EMPTY_HASH`
- It looks like `_opts` isn't used, so perhaps it can be replaced with `nil`

Reply:
Fixed in: 35b1090a4